### PR TITLE
Protocol tests

### PR DESCRIPTION
**Pull request by dlech**

### CODE REVIEW

**Comment by dlech** [comment]
This is revisiting #81 and #86. I have added some unit tests for the various race conditions when receiving messages with UNIX file descriptors. As seen in the comments and commit messages, there are some cases that cannot be fixed because of how twisted handles file descriptors - there is nothing to tie them to a specific message and since everything is async callbacks, the order in which callbacks are called is indeterminate.

I found a new bug that is fixed in 19bfecc. The new tests I added fail when this bug is not fixed.

Also, the existing `test_call_with_two_UNIX_FD_args` was failing for me locally even though it was not failing on travis. It is fixed by bf41eff.

**Comment by dlech** [comment]
I also filed an issue with twisted to see if there are any insights there. https://twistedmatrix.com/trac/ticket/9673

**Comment by dlech** [comment]
I just revised my fix for #81 and I'm much happier with it now. It covers one more theoretically possible race condition. And based on the feedback I got from twisted, the remaining theoretical race condition might not even be possible. (The real-world race condition that triggered these fixes in the first place is of course still fixed by this change as well.)

File: tests/test_protocol.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
+import struct
+
+from twisted.trial import unittest
+
+from txdbus import protocol
+
+
+class TestTransport():
```
Comment (WhyNotHugo) [inline tests/test_protocol.py]: Parenthesis are redundant (and not needed) for classes without any superclass.

**Comment by dlech** [comment]
Updated

**Pull request merged**